src/wp-admin/includes/deprecated.php

```diff
@@ -1589,3 +1589,31 @@ function image_attachment_fields_to_save( $post, $attachment ) {
 
 	return $post;
 }
+
+/**
+ * Resets global variables based on $_GET and $_POST.
+ *
+ * This function resets global variables based on the names passed
+ * in the $vars array to the value of $_POST[$var] or $_GET[$var] or ''
+ * if neither is defined.
+ *
+ * @since 2.0.0
+ * @deprecated 6.6.0
+ *
+ * @param array $vars An array of globals to reset.
+ */
+function wp_reset_vars( $vars ) {
```

Comment:
According to https://wpdirectory.net/search/01HWQEZAZ2VMB65S0FBX80HNZB there are some plugins using the function in their custom list tables. So maybe we shouldn't deprecate the function. We should still remove its usage in core though.